onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -169,6 +173,88 @@ TEST_F(CApiTest, DISABLED_custom_op) {
 }
 #endif
 
+struct OrtTensorDimensions : std::vector<int64_t> {
+  OrtTensorDimensions(OrtValue* value) {
+    OrtTensorTypeAndShapeInfo* info;
+    ORT_THROW_ON_ERROR(OrtGetTensorShapeAndType(value, &info));
+    auto dimensionCount = OrtGetNumOfDimensions(info);
+    resize(dimensionCount);
+    OrtGetDimensions(info, data(), dimensionCount);
+    OrtReleaseTensorTypeAndShapeInfo(info);
+  }
+
+  size_t ElementCount() const {
+    int64_t count = 1;
+    for (int i = 0; i < size(); i++)
+      count *= (*this)[i];
+    return count;
+  }
+};
+
+template <typename T, size_t N>
+constexpr size_t countof(T (&)[N]) { return N; }
+
+struct MyCustomKernel {
+  MyCustomKernel(OrtKernelInfo& /*info*/) {
+  }
+
+  void GetOutputShape(OrtValue** inputs, size_t /*input_count*/, size_t /*output_index*/, OrtTensorTypeAndShapeInfo* info) {
+    OrtTensorDimensions dimensions(inputs[0]);
+    ORT_THROW_ON_ERROR(OrtSetDims(info, dimensions.data(), dimensions.size()));
+  }
+
+  void Compute(OrtValue** inputs, size_t /*input_count*/, OrtValue** outputs, size_t /*output_count*/) {
+    const float* X;
```

Comment:
Given this is a test we should probably verify that input_count and output_count is as expected.